src/SFML/Window/Win32/WindowImplWin32.cpp

```diff
@@ -216,6 +212,12 @@ m_cursorGrabbed   (m_fullscreen)
     // Create the window
     m_handle = CreateWindowW(className, title.toWideString().c_str(), win32Style, left, top, width, height, NULL, NULL, GetModuleHandle(NULL), this);
 
+    // Remove all decorations if requested
+    if (style == Style::None)
+    {
+        SetWindowLong(m_handle, GWL_STYLE, 0);
```

Comment:
How did you find this solution? Is it documented?

Reply:
This basically sets the window to [`WS_OVERLAPPED`](https://msdn.microsoft.com/en-us/library/windows/desktop/ms632600(v=vs.85).aspx):

> The window is an overlapped window. An overlapped window has a title bar and a border. Same as the **WS_TILED** style.

This can't modify all styles after the window is created, so we're potentially just reverting one of the flags rather than everything and this works just because...

Reply:
How is it different from the style argument that we pass to `CreateWindowW`? Should be the same thing, and we already pass 0 to `CreateWindowW` (we just add `WS_VISIBLE` which has nothing to do with decorations).

Reply:
At window creation all components of `WS_OVERLAPPED` can be applied, so there might be happening more than when changing the flags only.

Reply:
The MSDN documentation of the `dwStyle` argument of `CreateWindow` and of the `GWL_STYLE` option of `SetWindowLong` both directly point to the window styles page, so (if the doc is correct) that should be the exact same thing.